lib/Parser/OglParser.php

```diff
@@ -151,72 +155,8 @@ public function parseStream($handle, $commit = null) {
-        foreach ($mesa->gl as $gl) {
-            $glName = (string) $gl['name'];
-            $glVersion = (string) $gl['version'];
 
-            $glSection = $this->matrix->getGlVersionByName($glName, $glVersion);
-            if (!$glSection) {
-                $glslName = (string) $gl->glsl['name'];
-                $glslVersion = (string) $gl->glsl['version'];
-
-                $glSection = new OglVersion($glName, $glVersion, $glslName, $glslVersion, $this->hints);
-                $this->matrix->addGlVersion($glSection);
-            }
-
-            // Add/merge new items in the matrix.
-            foreach ($gl->extension as $extension) {
-                // Create new extension.
-                $extName = (string) $extension['name'];
-                $extStatus = (string) $extension->mesa['status'];
-                $extHint = (string) $extension->mesa['hint'];
-                $newExtension = new OglExtension($extName, '', $this->hints, array());
-                $newExtension->setStatus($extStatus);
-                $newExtension->setHint($extHint);
-                foreach ($extension->supported->children() as $driver) {
-                    // Create new supported driver.
-                    $driverName = $driver->getName();
-                    $driverHint = (string) $driver['hint'];
-                    $driver = new OglSupportedDriver($driverName, $this->hints);
-                    $driver->setHint($driverHint);
-                    $newExtension->addSupportedDriver($driver);
-                }
-
-                // Add the extension.
-                $glExt = $glSection->addExtension($newExtension, $commit);
-                unset($extName, $extStatus, $extHint, $extDrivers, $newExtension);
-
-                foreach ($extension->subextension as $subextension) {
-                    // Create new sub-extension.
-                    $subExtName = (string) $subextension['name'];
-                    $subExtStatus = (string) $subextension->mesa['status'];
-                    $subExtHint = (string) $subextension->mesa['hint'];
-                    $newSubExtension = new OglExtension($subExtName, '', $this->hints, array());
-                    $newSubExtension->setStatus($subExtStatus);
-                    $newSubExtension->setHint($subExtHint);
-                    foreach ($subextension->supported->children() as $driver) {
-                        // Create new supported driver.
-                        $driverName = $driver->getName();
-                        $driverHint = (string) $driver['hint'];
-                        $driver = new OglSupportedDriver($driverName, $this->hints);
-                        $driver->setHint($driverHint);
-                        $newSubExtension->addSupportedDriver($driver);
-                    }
-
-                    // Add the sub-extension.
-                    $glSubExt = $glExt->addSubExtension($newSubExtension, $commit);
-                }
-            }
-        }
```

Comment:
This is replaced by a simple call to `OglMatrix::merge()` directly in Parse.php.